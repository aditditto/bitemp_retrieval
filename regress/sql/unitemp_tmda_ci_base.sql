DROP EXTENSION bitemp_retrieval;
CREATE EXTENSION bitemp_retrieval;
SET DATESTYLE TO 'ISO';
SET TIMEZONE TO 'Asia/Jakarta';

SELECT * FROM bitemporal_internal.ll_create_bitemporal_table(
    'public',
    'empl',
	$$n text not null, 
        cid int,
        d text not null,
        p text not null,
        h int not null,
        s int not null
	$$,
   'cid');

SELECT * FROM bitemporal_internal.ll_register_temporal_attribute_property('public', 'empl', 'h', 'malleable');

SELECT * FROM bitemporal_internal.temporal_attribute_properties;

-- r1
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Jan', 140, 'DB', 'P1', 2400, 1200$$,
    temporal_relationships.timeperiod('2003-01-01'::timestamptz, '2004-04-01'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

-- r2
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Jan', 163, 'DB', 'P1', 600, 1500$$,
    temporal_relationships.timeperiod('2004-07-01'::timestamptz, '2004-10-01'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

-- r3
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Ann', 141, 'DB', 'P2', 500, 700$$,
    temporal_relationships.timeperiod('2003-01-01'::timestamptz, '2003-06-01'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

-- r4
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Ann', 150, 'DB', 'P1', 1000, 800$$,
    temporal_relationships.timeperiod('2003-06-01'::timestamptz, '2004-04-01'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

-- r5
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Ann', 157, 'DB', 'P1', 600, 500$$,
    temporal_relationships.timeperiod('2004-01-01'::timestamptz, '2005-01-01'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

-- r6
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Sue', 142, 'DB', 'P2', 400, 800$$,
    temporal_relationships.timeperiod('2003-01-01'::timestamptz, '2003-11-01'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

-- r7
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Tom', 143, 'AI', 'P2', 1200, 2000$$,
    temporal_relationships.timeperiod('2003-04-01'::timestamptz, '2003-11-01'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

-- r8
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Tom', 153, 'AI', 'P1', 900, 1800$$,
    temporal_relationships.timeperiod('2004-01-01'::timestamptz, '2004-07-01'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

SELECT n, cid, d, p, h, s, effective FROM empl WHERE now() <@ asserted;

SELECT * FROM tmda_ci(
    'public',
    'empl',
    ARRAY['d'],
    ARRAY['sum', 'max'],
    ARRAY['h', 's'],
    ARRAY['sum_h', 'max_s']
) AS (
    d text,
    sum_h numeric,
    max_s numeric,
    effective tstzrange
) ORDER BY d DESC, lower(effective);


DROP TABLE public.empl;

SELECT * FROM bitemporal_internal.ll_create_bitemporal_table(
    'public',
    'empl',
	$$n text not null, 
        cid int,
        d text not null,
        p text not null,
        h int not null,
        s int not null
	$$,
   'cid');

-- r1
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Jan', 140, 'DB', 'P1', 2400, 1200$$,
    temporal_relationships.timeperiod('1970-01-01'::timestamptz, '1970-01-16'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

-- r2
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Jan', 163, 'DB', 'P1', 600, 1500$$,
    temporal_relationships.timeperiod('1970-01-19'::timestamptz, '1970-01-22'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

-- r3
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Ann', 141, 'DB', 'P2', 500, 700$$,
    temporal_relationships.timeperiod('1970-01-01'::timestamptz, '1970-01-06'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

-- r4
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Ann', 150, 'DB', 'P1', 1000, 800$$,
    temporal_relationships.timeperiod('1970-01-06'::timestamptz, '1970-01-16'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

-- r5
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Ann', 157, 'DB', 'P1', 600, 500$$,
    temporal_relationships.timeperiod('1970-01-13'::timestamptz, '1970-01-25'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

-- r6
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Sue', 142, 'DB', 'P2', 400, 800$$,
    temporal_relationships.timeperiod('1970-01-01'::timestamptz, '1970-01-11'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

-- r7
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Tom', 143, 'AI', 'P2', 1200, 2000$$,
    temporal_relationships.timeperiod('1970-01-04'::timestamptz, '1970-01-11'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

-- r8
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Tom', 153, 'AI', 'P1', 900, 1800$$,
    temporal_relationships.timeperiod('1970-01-13'::timestamptz, '1970-01-19'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);

SELECT n, cid, d, p, h, s, effective FROM empl WHERE now() <@ asserted ORDER BY LOWER(effective);

SELECT * FROM tmda_ci(
    'public',
    'empl',
    ARRAY['d'],
    ARRAY['sum', 'max'],
    ARRAY['h', 's'],
    ARRAY['sum_h', 'max_s']
) AS (
    d text,
    sum_h numeric,
    max_s numeric,
    effective tstzrange
) ORDER BY d DESC, lower(effective);

SELECT * FROM tmda_ci(
    'public',
    'empl',
    ARRAY['cid'],
    ARRAY['sum', 'max'],
    ARRAY['h', 's'],
    ARRAY['sum_h', 'max_s']
) AS (
    cid int,
    sum_h numeric,
    max_s numeric,
    effective tstzrange
) ORDER BY LOWER(effective);

DROP TABLE public.empl;