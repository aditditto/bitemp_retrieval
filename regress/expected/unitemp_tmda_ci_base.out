DROP EXTENSION bitemp_retrieval;
CREATE EXTENSION bitemp_retrieval;
SET DATESTYLE TO 'ISO';
SET TIMEZONE TO 'Asia/Jakarta';
SELECT * FROM bitemporal_internal.ll_create_bitemporal_table(
    'public',
    'empl',
	$$n text not null, 
        cid int,
        d text not null,
        p text not null,
        h int not null,
        s int not null
	$$,
   'cid');
 ll_create_bitemporal_table 
----------------------------
 t
(1 row)

SELECT * FROM bitemporal_internal.ll_register_temporal_attribute_property('public', 'empl', 'h', 'malleable');
 ll_register_temporal_attribute_property 
-----------------------------------------
                                       1
(1 row)

SELECT * FROM bitemporal_internal.temporal_attribute_properties;
 schema_name | table_name | attr_name | attr_property 
-------------+------------+-----------+---------------
 public      | empl       | h         | malleable
(1 row)

-- r1
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Jan', 140, 'DB', 'P1', 2400, 1200$$,
    temporal_relationships.timeperiod('2003-01-01'::timestamptz, '2004-04-01'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

-- r2
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Jan', 163, 'DB', 'P1', 600, 1500$$,
    temporal_relationships.timeperiod('2004-07-01'::timestamptz, '2004-10-01'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

-- r3
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Ann', 141, 'DB', 'P2', 500, 700$$,
    temporal_relationships.timeperiod('2003-01-01'::timestamptz, '2003-06-01'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

-- r4
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Ann', 150, 'DB', 'P1', 1000, 800$$,
    temporal_relationships.timeperiod('2003-06-01'::timestamptz, '2004-04-01'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

-- r5
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Ann', 157, 'DB', 'P1', 600, 500$$,
    temporal_relationships.timeperiod('2004-01-01'::timestamptz, '2005-01-01'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

-- r6
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Sue', 142, 'DB', 'P2', 400, 800$$,
    temporal_relationships.timeperiod('2003-01-01'::timestamptz, '2003-11-01'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

-- r7
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Tom', 143, 'AI', 'P2', 1200, 2000$$,
    temporal_relationships.timeperiod('2003-04-01'::timestamptz, '2003-11-01'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

-- r8
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Tom', 153, 'AI', 'P1', 900, 1800$$,
    temporal_relationships.timeperiod('2004-01-01'::timestamptz, '2004-07-01'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

SELECT n, cid, d, p, h, s, effective FROM empl WHERE now() <@ asserted;
  n  | cid | d  | p  |  h   |  s   |                      effective                      
-----+-----+----+----+------+------+-----------------------------------------------------
 Jan | 140 | DB | P1 | 2400 | 1200 | ["2003-01-01 00:00:00+07","2004-04-01 00:00:00+07")
 Jan | 163 | DB | P1 |  600 | 1500 | ["2004-07-01 00:00:00+07","2004-10-01 00:00:00+07")
 Ann | 141 | DB | P2 |  500 |  700 | ["2003-01-01 00:00:00+07","2003-06-01 00:00:00+07")
 Ann | 150 | DB | P1 | 1000 |  800 | ["2003-06-01 00:00:00+07","2004-04-01 00:00:00+07")
 Ann | 157 | DB | P1 |  600 |  500 | ["2004-01-01 00:00:00+07","2005-01-01 00:00:00+07")
 Sue | 142 | DB | P2 |  400 |  800 | ["2003-01-01 00:00:00+07","2003-11-01 00:00:00+07")
 Tom | 143 | AI | P2 | 1200 | 2000 | ["2003-04-01 00:00:00+07","2003-11-01 00:00:00+07")
 Tom | 153 | AI | P1 |  900 | 1800 | ["2004-01-01 00:00:00+07","2004-07-01 00:00:00+07")
(8 rows)

SELECT * FROM tmda_ci(
    'public',
    'empl',
    ARRAY['d'],
    ARRAY['sum', 'max'],
    ARRAY['h', 's'],
    ARRAY['sum_h', 'max_s']
) AS (
    d text,
    sum_h numeric,
    max_s numeric,
    effective tstzrange
) ORDER BY d DESC, lower(effective);
 d  |      sum_h       | max_s |                      effective                      
----+------------------+-------+-----------------------------------------------------
 DB | 1493.42105263158 |  1200 | ["2003-01-01 00:00:00+07","2003-06-01 00:00:00+07")
 DB | 1508.21829163072 |  1200 | ["2003-06-01 00:00:00+07","2003-11-01 00:00:00+07")
 DB | 521.052631578947 |  1200 | ["2003-11-01 00:00:00+07","2004-01-01 00:00:00+07")
 DB |  926.48835202761 |  1200 | ["2004-01-01 00:00:00+07","2004-04-01 00:00:00+07")
 DB | 149.180327868852 |   500 | ["2004-04-01 00:00:00+07","2004-07-01 00:00:00+07")
 DB | 750.819672131148 |  1500 | ["2004-07-01 00:00:00+07","2004-10-01 00:00:00+07")
 DB | 150.819672131148 |   500 | ["2004-10-01 00:00:00+07","2005-01-01 00:00:00+07")
 AI |             1200 |  2000 | ["2003-04-01 00:00:00+07","2003-11-01 00:00:00+07")
 AI |              900 |  1800 | ["2004-01-01 00:00:00+07","2004-07-01 00:00:00+07")
(9 rows)

DROP TABLE public.empl;
SELECT * FROM bitemporal_internal.ll_create_bitemporal_table(
    'public',
    'empl',
	$$n text not null, 
        cid int,
        d text not null,
        p text not null,
        h int not null,
        s int not null
	$$,
   'cid');
 ll_create_bitemporal_table 
----------------------------
 t
(1 row)

-- r1
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Jan', 140, 'DB', 'P1', 2400, 1200$$,
    temporal_relationships.timeperiod('1970-01-01'::timestamptz, '1970-01-16'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

-- r2
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Jan', 163, 'DB', 'P1', 600, 1500$$,
    temporal_relationships.timeperiod('1970-01-19'::timestamptz, '1970-01-22'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

-- r3
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Ann', 141, 'DB', 'P2', 500, 700$$,
    temporal_relationships.timeperiod('1970-01-01'::timestamptz, '1970-01-06'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

-- r4
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Ann', 150, 'DB', 'P1', 1000, 800$$,
    temporal_relationships.timeperiod('1970-01-06'::timestamptz, '1970-01-16'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

-- r5
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Ann', 157, 'DB', 'P1', 600, 500$$,
    temporal_relationships.timeperiod('1970-01-13'::timestamptz, '1970-01-25'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

-- r6
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Sue', 142, 'DB', 'P2', 400, 800$$,
    temporal_relationships.timeperiod('1970-01-01'::timestamptz, '1970-01-11'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

-- r7
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Tom', 143, 'AI', 'P2', 1200, 2000$$,
    temporal_relationships.timeperiod('1970-01-04'::timestamptz, '1970-01-11'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

-- r8
SELECT * FROM bitemporal_internal.ll_bitemporal_insert(
    'public.empl',
    $$n, cid, d, p, h, s$$,
    $$'Tom', 153, 'AI', 'P1', 900, 1800$$,
    temporal_relationships.timeperiod('1970-01-13'::timestamptz, '1970-01-19'::timestamptz), --effective
    temporal_relationships.timeperiod(now(), 'infinity') --asserted
);
 ll_bitemporal_insert 
----------------------
                    1
(1 row)

SELECT n, cid, d, p, h, s, effective FROM empl WHERE now() <@ asserted ORDER BY LOWER(effective);
  n  | cid | d  | p  |  h   |  s   |                      effective                      
-----+-----+----+----+------+------+-----------------------------------------------------
 Sue | 142 | DB | P2 |  400 |  800 | ["1970-01-01 00:00:00+07","1970-01-11 00:00:00+07")
 Ann | 141 | DB | P2 |  500 |  700 | ["1970-01-01 00:00:00+07","1970-01-06 00:00:00+07")
 Jan | 140 | DB | P1 | 2400 | 1200 | ["1970-01-01 00:00:00+07","1970-01-16 00:00:00+07")
 Tom | 143 | AI | P2 | 1200 | 2000 | ["1970-01-04 00:00:00+07","1970-01-11 00:00:00+07")
 Ann | 150 | DB | P1 | 1000 |  800 | ["1970-01-06 00:00:00+07","1970-01-16 00:00:00+07")
 Tom | 153 | AI | P1 |  900 | 1800 | ["1970-01-13 00:00:00+07","1970-01-19 00:00:00+07")
 Ann | 157 | DB | P1 |  600 |  500 | ["1970-01-13 00:00:00+07","1970-01-25 00:00:00+07")
 Jan | 163 | DB | P1 |  600 | 1500 | ["1970-01-19 00:00:00+07","1970-01-22 00:00:00+07")
(8 rows)

SELECT * FROM tmda_ci(
    'public',
    'empl',
    ARRAY['d'],
    ARRAY['sum', 'max'],
    ARRAY['h', 's'],
    ARRAY['sum_h', 'max_s']
) AS (
    d text,
    sum_h numeric,
    max_s numeric,
    effective tstzrange
) ORDER BY d DESC, lower(effective);
 d  | sum_h | max_s |                      effective                      
----+-------+-------+-----------------------------------------------------
 DB |  1500 |  1200 | ["1970-01-01 00:00:00+07","1970-01-06 00:00:00+07")
 DB |  1500 |  1200 | ["1970-01-06 00:00:00+07","1970-01-11 00:00:00+07")
 DB |   520 |  1200 | ["1970-01-11 00:00:00+07","1970-01-13 00:00:00+07")
 DB |   930 |  1200 | ["1970-01-13 00:00:00+07","1970-01-16 00:00:00+07")
 DB |   150 |   500 | ["1970-01-16 00:00:00+07","1970-01-19 00:00:00+07")
 DB |   750 |  1500 | ["1970-01-19 00:00:00+07","1970-01-22 00:00:00+07")
 DB |   150 |   500 | ["1970-01-22 00:00:00+07","1970-01-25 00:00:00+07")
 AI |  1200 |  2000 | ["1970-01-04 00:00:00+07","1970-01-11 00:00:00+07")
 AI |   900 |  1800 | ["1970-01-13 00:00:00+07","1970-01-19 00:00:00+07")
(9 rows)

SELECT * FROM tmda_ci(
    'public',
    'empl',
    ARRAY['cid'],
    ARRAY['sum', 'max'],
    ARRAY['h', 's'],
    ARRAY['sum_h', 'max_s']
) AS (
    cid int,
    sum_h numeric,
    max_s numeric,
    effective tstzrange
) ORDER BY LOWER(effective);
 cid | sum_h | max_s |                      effective                      
-----+-------+-------+-----------------------------------------------------
 142 |   400 |   800 | ["1970-01-01 00:00:00+07","1970-01-11 00:00:00+07")
 141 |   500 |   700 | ["1970-01-01 00:00:00+07","1970-01-06 00:00:00+07")
 140 |  2400 |  1200 | ["1970-01-01 00:00:00+07","1970-01-16 00:00:00+07")
 143 |  1200 |  2000 | ["1970-01-04 00:00:00+07","1970-01-11 00:00:00+07")
 150 |  1000 |   800 | ["1970-01-06 00:00:00+07","1970-01-16 00:00:00+07")
 153 |   900 |  1800 | ["1970-01-13 00:00:00+07","1970-01-19 00:00:00+07")
 157 |   600 |   500 | ["1970-01-13 00:00:00+07","1970-01-25 00:00:00+07")
 163 |   600 |  1500 | ["1970-01-19 00:00:00+07","1970-01-22 00:00:00+07")
(8 rows)

